pkg/platform/provider/baremetal/machine/create.go

```diff
@@ -386,7 +451,10 @@ func (p *Provider) EnsureKubeadm(ctx context.Context, machine *platformv1.Machin
 		return err
 	}
 
-	err = kubeadm.Install(machineSSH, cluster.Spec.Version)
+	option := &kubeadm.Option{
+		RuntimeType: cluster.Spec.Features.ContainerRuntime,
+	}
+	err = kubeadm.Install(machineSSH, cluster.Spec.Version, option)
```

Comment:
why use option to this func, not cluster.Spec.Features.ContainerRuntim

Reply:
Remove the cluster.Spec.Version into option.